### PR TITLE
Shell Recorder: Minor Enhancements

### PR DESCRIPTION
**Pull request by sanssecours**
# Purpose

This PR should fix issue #1936. It also contains other minor enhancements for the (Markdown) Shell Recorder.

# Checklist

- [x] I checked all commit messages. The issue references should be correct.
- [x] I ran all tests locally and everything went fine.
- [x] This PR contains an updated version of the release notes.

### CODE REVIEW

File: tests/shell/shell_recorder/shell_recorder.sh [hunk 1]
```diff
@@ -57,12 +57,12 @@ execute()
 
 	if [ -n "$RETCMP" ];
 	then
-		nbTest=$(( nbTest + 1 ))
+		tests=$(( tests + 1 ))
```
Comment (markus2330) [inline tests/shell/shell_recorder/shell_recorder.sh]: The name "nbTest" is also used in run_all and the cframework. errors and tests are really generic words which I would rather avoid as variable name. (Makes grepping and search&replace more difficult.)

**Comment by markus2330** [comment]
Thank you for working on this! The shell recorder definitely needs some love!

**Comment by sanssecours** [comment]
jenkins build all please

**Comment by sanssecours** [comment]
jenkins build haskell please

**Comment by sanssecours** [comment]
jenkins build mingw64 please

**Comment by sanssecours** [comment]
jenkins build mingw64 please

**Comment by sanssecours** [comment]
jenkins build all please

**Comment by markus2330** [comment]
Thank you, great work!

**Pull request merged**